pkg/api/v1alpha1/cluster.go

```diff
@@ -213,6 +226,19 @@ func GetAndValidateClusterConfig(fileName string) (*Cluster, error) {
 		return nil, err
 	}
 
+	if clusterConfig.Spec.GitOpsRef != nil {
```

Comment:
you dont validate the fluxConfig here, instead u can do it similar to https://github.com/aws/eks-anywhere/blob/main/pkg/cluster/gitops.go

`GetAndValidateClusterConfig` only validate cluster object itself (including the refs)

Reply:
Agreed, here we only validate if the ref has all the values or something from the perspective of the cluster object.

Reply:
For sure. Once #2034 merges, I'll refactor this PR to add the validations in the proper place